GitUI/UserControls/RevisionGrid/Graph/RevisionGraphRow.cs

```diff
@@ -29,14 +32,25 @@ public RevisionGraphRow(RevisionGraphRevision revision, IReadOnlyList<RevisionGr
         public IReadOnlyList<RevisionGraphSegment> Segments { get; }
 
         // This dictonary contains a cached list of all segments and the lane index the segment is in for this row.
-        private IReadOnlyDictionary<RevisionGraphSegment, int>? _segmentLanes;
+        private IDictionary<RevisionGraphSegment, int>? _segmentLanes;
 
         // The cached lanecount
         private int _laneCount;
 
         // The cached revisionlane
         private int _revisionLane;
 
+        public void MoveLanesRight(int fromLane)
+        {
+            ++_laneCount;
+            Validates.NotNull(_segmentLanes);
+            RevisionGraphSegment[] segmentsToBeMoved = _segmentLanes.Where(keyValue => keyValue.Value >= fromLane).Select(keyValue => keyValue.Key).ToArray();
```

Comment:
```suggestion
            RevisionGraphSegment[] segmentsToBeMoved = _segmentLanes.Where(keyValue => keyValue.Value >= fromLane).Select(keyValue => keyValue.Key);
```

Reply:
This is not an iterator of the C++ std lib, which allows the modification of values.
Without, it throws `InvalidOperationException` "Collection has changed".